test/github/graphql.test.js

```diff
@@ -0,0 +1,80 @@
+const GitHub = require('../../lib/github')
+const nock = require('nock')
+const Bottleneck = require('bottleneck')
+
+describe('github/graphql', () => {
+  let github
+
+  // Expect there are no more pending nock requests
+  beforeEach(async () => nock.cleanAll())
+  afterEach(() => expect(nock.pendingMocks()).toEqual([]))
+
+  beforeEach(() => {
+    const logger = {
+      debug: jest.fn(),
+      trace: jest.fn()
+    }
+
+    // Set a shorter limiter, otherwise tests are _slow_
+    const limiter = new Bottleneck({ maxConcurrent: 1, minTime: 1 })
+
+    github = new GitHub({ logger, limiter })
+  })
+
+  describe('query', () => {
+    let query
+    let data
+
+    test('makes a graphql query', async () => {
+      query = 'query { viewer { login } }'
+      data = { viewer: { login: 'bkeepers' } }
+
+      nock('https://api.github.com', {
+        reqheaders: { 'content-type': 'application/json' }
+      }).post('/graphql', {query})
+        .reply(200, { data })
+
+      expect(await github.query(query)).toEqual(data)
+    })
+
+    test('makes a graphql query with variables', async () => {
+      const variables = {owner: 'probot', repo: 'test'}
+
+      nock('https://api.github.com', {
+        reqheaders: { 'content-type': 'application/json' }
+      }).post('/graphql', {query, variables})
+        .reply(200, { data })
+
+      expect(await github.query(query, variables)).toEqual(data)
+    })
+
+    test('uses authentication', async () => {
+      github.authenticate({type: 'token', token: 'testing'})
+
+      nock('https://api.github.com', {
+        reqheaders: { authorization: 'token testing' }
+      }).post('/graphql', {query})
+        .reply(200, { data })
+
+      await github.query(query)
+    })
+
+    test('allows custom headers', async () => {
+      nock('https://api.github.com', {
+        reqheaders: { 'foo': 'bar' }
+      }).post('/graphql', {query})
+        .reply(200, { data })
+
+      await github.query(query, undefined, {foo: 'bar'})
```

Comment:
>What would you all think about having the query be the first argument, and then variables and headers as named arguments?

That looks good to me :)

Reply:
Ok, now I'm second guessing this. Since variables is likely to be used _way_ more often than headers, it feels awkward to always require variables to always be namespaced. I'm going to leave this as-is for now and we can iterate on it in a future PR.